user_sessions/views.py

```diff
@@ -0,0 +1,46 @@
+from django.contrib.auth import authenticate, login, logout
+from django.http import HttpResponseRedirect
+from django.shortcuts import render
+from django.urls import reverse
+from django.views import View
+
+from user_sessions.forms import LoginForm
+
+
+class LoginView(View):
+    form_class = LoginForm
+    template_name = 'user_sessions/login.html'
+
+    def get(self, request):
+        form = self.form_class()
+        return render(request, self.template_name, {'form': form})
+
+    def post(self, request):
+        form = self.form_class(request.POST)
+        if form.is_valid():
+            un = form.cleaned_data['username']
+            pw = form.cleaned_data['password']
+            user = authenticate(username=un, password=pw)
+            if user:
+                login(request, user)
+                return HttpResponseRedirect(reverse('contacts:contact_list'))
+            else:
+                return render(request, self.template_name, {'form': form, 'error_message':
+                    "Invalid Username or Password. Please try again. "})
+        else:
+            return render(request, self.template_name, {'form': form})
+
+
+class HomePageView(View):
+    template_name = 'phone_directory/home.html'
+
+    def get(self, request):
+        return render(request, self.template_name)
+
+
+class LogoutView(View):
+    template_name = 'phone_directory/home.html'
```

Comment:
this template is associated with HomePageView. we should not render it directly from here, instead redirect to home view after logout.